File: src/main/java/nl/uu/fi/dwo/lms/gwtclient/gwt/jsdisplays/JsAccountDisplay.java
package nl.uu.fi.dwo.lms.gwtclient.gwt.jsdisplays;

import jsinterop.annotations.JsPackage;
import jsinterop.annotations.JsType;

/**
 * AccountDisplay UI interface. the interface should be
 * available as a JavaScript object named "jsAccountDisplay".
 * Callback from the MainDisplay UI to the presenter occurs
 * via 
 * 
 * There are two relevant presenter functions.
 * 
 * To save account data:
 * 
 * public void saveUser(String givenName, String insertion, String familyName, 
 *   String email, String curPassword, String newPassword, String newPasswordAgain);
 * 
 * and to reset the data in the Account display.
 * 
 * public void updateUserDataInView();
 * 
 * @author G.A.J. van der Plas
 */
@JsType(isNative = true, name = "jsAccountDisplay", namespace = JsPackage.GLOBAL)
public class JsAccountDisplay{
    /** Clears all the account fields of their values. */
    public static native void clear();
    /** Clears the password field of its value. */
    public static native void init();
    /** Clears username and password fields of values. */
    public static native void updateView(String username, String firstName, String insertion, String familyName, String email);
}

File: war/js-plainui/jsAccountDisplay.js
function AccountDisplay() {
    var username = "";
    var firstName = "";
    var insertion = "";
    var familyName = "";
    var email = "";
    var password = "";
    var newPassword = "";
    var newPasswordAgain = "";

    this.clear = function () {
        username = "";
        firstName = "";
        insertion = "";
        familyName = "";
        password = "";
        email = "";
        console.log("cleared account fields");
    }

    this.init = function () {
        password = "";
        console.log("intialized account display, cleared password.");
    }
    this.updateView = function (u, f, i, n, e) {
        username = u;
        firstName = f;
        insertion = i;
        familyName = n;
        email = e;
        console.log("updated username: " + u + ", firstname:" + f + ", insertion:" + i + ", familyname:" + n + ", email: " + e + ".");
    }
    this.showState = function (){
        console.log("state-info| username: " + u + ", firstname:" + f + ", insertion:" + i + ", familyname:" + n + ", email: " + e + ".");
    }
    
}
var jsAccountDisplay = new AccountDisplay();
console.log("constructed jsAccountDisplay");

